### ISSUE TITLE
bug: Unable to forward SNS message to SQS FIFO: MissingParameter MessageGroupId

### ISSUE THREAD

**Issue by ndakic** [localstack/localstack#5132]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

For some reason Localstack is not able to forward SNS Message to SQS FIFO.
I tried to explicitly set MessageGroupId attribute in code, but it's unable to forward message from SNS to SQS.
I am not sure is this bug or it is not supported by Localstack yet, or I am doing something wrong.

#### Localstack stacktrace:
```
INFO:localstack.services.sns.sns_listener: Unable to forward SNS message to SQS: An error occurred (MissingParameter) when calling the SendMessage operation: The request must contain the parameter MessageGroupId. Traceback (most recent call last):
localstack  |   File "/opt/code/localstack/localstack/services/sns/sns_listener.py", line 467, in message_to_subscriber
localstack  |     sqs_client.send_message(
localstack  |   File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 391, in _api_call
localstack  |     return self._make_api_call(operation_name, kwargs)
localstack  |   File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 719, in _make_api_call
localstack  |     raise error_class(parsed_response, operation_name)
localstack  | botocore.exceptions.ClientError: An error occurred (MissingParameter) when calling the SendMessage operation: The request must contain the parameter MessageGroupId.
```

### Expected Behavior

I want to be able to pull message from SQS when I publish message to SNS.

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker compose up localstack

docker-compose.yml
```
version: "3"
services:
  localstack:
    image: localstack/localstack:latest
    container_name: localstack
    ports:
      - "4566-4599:4566-4599"
```
#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    awslocal sqs create-queue --queue-name myapp-input-local.fifo --attributes "FifoQueue=true,ContentBasedDeduplication=false"

    awslocal sns create-topic --name testapp-service-output-local

    awslocal sns subscribe --topic-arn arn:aws:sns:us-east-1:000000000000:testapp-service-output-local --protocol sqs --notification-endpoint http://localhost:4566/000000000000/myapp-input-local.fifo --attributes "RawMessageDelivery=True"`

#### python code

```
sns = boto3.client('sns',
                   aws_access_key_id=e_config['aws_access_key_id'],
                   aws_secret_access_key=e_config['aws_secret_access_key'],
                   region_name=e_config['aws_region'],
                   endpoint_url=e_config['aws_endpoint_url'])

sns.publish(Message=message, TopicArn=e_config['aws_response_arn'], MessageAttributes=message_attributes)
```

### Environment

```markdown
- OS: ppc64le linux (Localstack is based on python:3.8.12-slim-buster image)
- LocalStack: 0.13.0.7
```

**Comment by codingtim** [comment]
We seem to have a related issue. 
We send a message to SNS with MessageGroupId and should be received on subscribed SQS. 
In version 0.13.0 this works. 
In version 0.13.1 the message does not end up on the SQS. After removing the MessageGroupId from the SNS request, the messages arrives on the SQS. 
Something seems off with SNS/SQS subscription.

**Comment by pioneer2k** [comment]
The same for me. With version 0.13.1 and 0.13.2 (Docker) the communication between SNS and SQS in all our tests is totally broken. With 0.13.0 everything is fine. When I remove the messageGroupId everything works again, but we need the messageGroupId.

**Comment by ndakic** [comment]
@codingtim @pioneer2k yeah, it's working with 0.13.0 version. For now, that's okay to me. Thank you guys

**Comment by pinzon** [comment]
Hi, after replicating the behaviour on AWS I found out that Fifo Queues cannot be subscribed to standard SNS topics, they must be Fifo Topics. 

One should get this error when trying it:
```An error occurred (InvalidParameter) when calling the Subscribe operation: Invalid parameter: Invalid parameter: Endpoint Reason: FIFO SQS Queues can not be subscribed to standard SNS topics```

Fifo Topics force messages to have MessageGroupId which is the parameter needed for Fifo SQS queues.  That's why you're getting that error message.

**Comment by pioneer2k** [comment]
@pinzon your last comment applies to the original issue, but my issue with versions 0.13.1 and 0.13.2 has nothing to do with FIFO topics. We are not using FIFO topics. Do you think that is worth opening another issue?

**Comment by pinzon** [comment]
@pioneer2k  Yes. please open a new issue for the situation you describe, having an specific Issue for each PR makes easier to understand of what they fix and easy to reproduce. 

Said that, you could wait for the PR to be merged, it includes a test that makes sure that a sns message with MessageGroupId is being delivered to the sqs queue.

**Comment by whummer** [comment]
Thanks again @pioneer2k @pinzon @ndakic . The original issue described in this thread should be (hopefully) fixed in #5132 . Can you please pull the `latest` Docker image and give it another try? Please re-open this issue, or create a new issue with detailed steps to reproduce, in case the problem persists. Thanks for your help!

**Comment by codingtim** [comment]
Running with latest our tests with messageGroupId work again :+1:
